src/BufferManager.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file BufferManager.cpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#include "trigger/BufferManager.hpp"
+
+#include <iostream>
+
+namespace dunedaq::trigger {
+
+BufferManager::BufferManager(long unsigned int buffer_size)
+  : m_buffer_max_size(buffer_size)
+  , m_buffer_earliest_start_time(0)
+  , m_buffer_latest_end_time(0)
+{
+
+}
+
+BufferManager::~BufferManager()
+{
+
+}
+
+bool
+BufferManager::add(trigger::TPSet& tps)
+{
+  if(m_tpset_buffer.size() >= m_buffer_max_size) //delete oldest TPSet if buffer full -> circular buffer
+  {
+    auto firstIt = m_tpset_buffer.begin();
+    m_tpset_buffer.erase(firstIt);
```

Comment:
I think you need to update `buffer_earliest_start_time` here, right?

Reply:
Indeed! Thanks! Fixed.